src/service/LinkService.ts

```diff
@@ -0,0 +1,152 @@
+import Logger from '../logger/Logger';
+import LoggerFactory from '../logger/LoggerFactory';
+import { LogType } from '../logger/LogType';
+import {
+    Account,
+    Deadline,
+    LinkAction,
+    RepositoryFactoryHttp,
+    Transaction,
+    TransactionService,
+    UInt64,
+    VotingKeyLinkTransaction,
+    VrfKeyLinkTransaction,
+} from 'symbol-sdk';
+import { BootstrapUtils } from './BootstrapUtils';
+import * as _ from 'lodash';
+import { Addresses, ConfigPreset, NodeAccount } from '../model';
+import { catchError, map, mergeMap, toArray } from 'rxjs/operators';
+import { fromArray } from 'rxjs/internal/observable/fromArray';
+import { EMPTY, Observable, of } from 'rxjs';
+
+/**
+ * params necessary to announce link transactions network.
+ */
+export type LinkParams = { target: string; url: string; maxFee: number };
+
+const logger: Logger = LoggerFactory.getLogger(LogType.System);
+
+export class LinkService {
+    public static readonly defaultParams: LinkParams = { target: 'target', url: 'http://localhost:3000', maxFee: 100000 };
+
+    constructor(protected readonly params: LinkParams) {}
+
+    public async run(passedPresetData?: ConfigPreset | undefined, passedAddresses?: Addresses | undefined): Promise<void> {
+        const presetData = passedPresetData ?? BootstrapUtils.loadExistingPresetData(this.params.target);
+        const addresses = passedAddresses ?? BootstrapUtils.loadExistingAddresses(this.params.target);
+
+        const url = this.params.url.replace(/\/$/, '');
+        logger.info(`Linking nodes using network url ${url}. Max Fee ${this.params.maxFee}`);
+        const repositoryFactory = new RepositoryFactoryHttp(url);
+
+        const generationHash = await repositoryFactory.getGenerationHash().toPromise();
+        if (generationHash !== presetData.nemesisGenerationHashSeed) {
+            throw new Error(
+                `You are connecting to the wrong network. Expected generation hash is ${presetData.nemesisGenerationHashSeed} but got ${generationHash}`,
+            );
+        }
+
+        const transactionNodes: { node: NodeAccount; transactions: Transaction[] }[] = _.flatMap(addresses.nodes || []).map((node) => {
+            const transactions = [];
+            const account = Account.createFromPrivateKey(node.signing.privateKey, presetData.networkType);
+            if (node.voting) {
+                const votingPublicKey = BootstrapUtils.createVotingKey(node.voting.publicKey);
+                logger.info(
+                    `Creating VotingKeyLinkTransaction - node: ${node.name}, signer public key: ${account.publicKey}, voting public key: ${votingPublicKey}`,
+                );
+                transactions.push(
+                    VotingKeyLinkTransaction.create(
+                        Deadline.create(),
+                        votingPublicKey,
+                        1,
+                        26280,
```

Comment:
Not at an expert on these values.

I think we could allow users to provide custom values via presets. Then bootstrap will use those values to generate the tree files via catapult.tools.votingkey and to generate the right values in the VotingKeyLinkTransaction. This could be a future PR.

Reply:
Added preset params defaulted to the default ones:)